### PR TITLE
[flutter_tools] standardize patterns for integration.shard

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Integration tests must only go through the real file system/process manager/platform. The global indirection makes this code harder to understand than if it directly referred to the concrete instances that are being used.

Update the integration shard to use a const instance of a LocalFIleSystem, LocalProcessManager, and LocalPlatform. Remove global usage and apply testWithoutContext.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/forbidden_imports_test.dart [hunk 1]
```diff
@@ -2,22 +2,24 @@
...
 // found in the LICENSE file.
 
-import 'package:flutter_tools/src/base/file_system.dart';
-import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:file/file.dart';
+import 'package:file/local.dart';
 
 import '../src/common.dart';
 
+const FileSystem fileSystem = LocalFileSystem();
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: Doesn't have to be in this PR, but should this be moved to the integration shard?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: I wasn't sure. It is definitely touching the real filesystem, but it is also more of an analysis check - I could see wanting it to remain fast.

File: packages/flutter_tools/test/general.shard/forbidden_imports_test.dart [hunk 2]
```diff
@@ -136,18 +138,20 @@ void main() {
 
   test('no unauthorized imports of package:file/local.dart', () {
     final List<String> allowedPath = <String>[
-      globals.fs.path.join(flutterTools, 'lib', 'src', 'base', 'file_system.dart'),
+      fileSystem.path.join(flutterTools, 'test', 'general.shard', 'forbidden_imports_test.dart'),
+      fileSystem.path.join(flutterTools, 'test', 'integration.shard', 'test_utils.dart'),
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:142]: would it be possible to allow the entire `//test/integration.shard/` dir?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:142]: we could - this sort of enforces using the existing test_utils, which could be helpful if we do want to change the file system in the future

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:142]: oh yeah, makes sense

File: packages/flutter_tools/test/general.shard/forbidden_imports_test.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: Does it run in parallel with other general shard tests?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: yes ... hopefully those aren't also writing to the filesytem...hmm

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: we should probably just move this?

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: i vote move them just to be safe and rule out the potential flaking.

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/forbidden_imports_test.dart:10]: done!

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
This does not affect google3 unless they are trying to run our integration tests.

File: packages/flutter_tools/test/integration.shard/test_utils.dart [hunk 3]
```diff
@@ -4,36 +4,47 @@
...
+import 'package:file/file.dart';
+import 'package:file/local.dart';
 import 'package:flutter_tools/src/base/io.dart';
-import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:process/process.dart';
 
 import '../src/common.dart';
 
+/// The [FileSystem] for the integration test environment.
```
Comment (zanderso) [inline packages/flutter_tools/test/integration.shard/test_utils.dart:15]: How about a test in forbidden_imports_test.dart that checks that this file isn't imported outside of the integration.shard subdir?

**Review by zanderso — COMMENTED** [review]
sorry for the late comment.

**Pull request merged**